Always prefix .pb.dart imports in .pbserver, .pb.json, .pbgrpc files.

nichite opened the pull request:
A lack of prefixing has led to internal issues where invalid code is generated (names generated in the main .pb.dart file or its imports collide with names in the .pbserver, .pb.json, or .pbgrpc generated files.

rakudrama commented on protoc_plugin/lib/file_generator.dart:
```diff
@@ -570,12 +570,20 @@ class FileGenerator extends ProtobufContainer {
   void _writeImport(IndentingWriter out, OutputConfiguration config,
-      FileGenerator target, String extension) {
+      FileGenerator target, String extension,
+      [bool forMainFile = false]) {
```
I'd use a named optional parameter - it is easier to understand what 'true' means at the call site.

nichite replied:
I ended up just taking the parameter out and always prefixing as per the other suggestion.

rakudrama commented on protoc_plugin/lib/file_generator.dart:
```diff
@@ -570,12 +570,20 @@ class FileGenerator extends ProtobufContainer {
   /// Writes an import of a .dart file corresponding to a .proto file.
   /// (Possibly the same .proto file.)
+  ///
+  /// When generating the main file (if [forMainFile] is true), all imports
```
What do you want to omit the import prefix?
Can the logic be simplified by never not using a prefix?

sigurdm replied:
I am also a bit confused. I think we might as well always prefix...

nichite replied:
Done, might as well always prefix as you both suggested.

nichite commented:
So everything's green again--to clarify on the parameter: this doesn't decide whether or not to prefix `pb.dart` file imports--it always prefixed them. However, I decided it was doing the same thing as a basic check against the file extension, so I removed it in `file_generator.dart` and just check if it's a `pb.dart` file.

The other check in `_getDartClassName` remains (in `service_generator.dart`). The whole reason for this PR is essentially: the `protoFileUri != target.protoFileUri` check doesn't always do what we want it to. It evaluates to true when generating, say, `hello.pb.dart` from `hello.proto`, where we know those generated Dart classes will be in the same file, but it also evaluates to true for `hello.pbserver.dart`, `hello.pbgrpc.dart`, etc. In those cases collisions can occur from the generated class, so best to always prefix `pb.dart` imports.

The one exception would be `client_generator.dart`--the generated client API doesn't get its own file (unlike the server), but it uses the same `_getDartClassName` method as the server file, so it uses the param to say "hey, I'm in the same main pb.dart file where all these classes are used." I added some more comments to make that more clear.
